training/training.module

```diff
@@ -0,0 +1,378 @@
+  );
+  $menu['training/theme/access'] = array(
+    'title' => 'Access Theme',
+    'description' => 'This page is Access Theme page',
+    'page callback' => 'training_access_callback',
+    'access callback' => TRUE,
+    'file' => 'training.pages.inc',
+  );
+  $menu['training/form'] = array(
+    'title' => 'Form Example',
+    'description' => 'This page is Form Example',
+    'page callback' => 'training_example_form_callback',
+    'page arguments' => array('training_google_form', ),
+    'access callback' => TRUE,
+    'file' => 'training.pages.inc',
+  );
+  return $menu;
+}
+
+/**
+ * Check access for page Custom Access Menu Item.
+ * $user - a global variable keeps the user basic information about the current user.
+ * $role_name - param contents role authenticated user
+ * @return bool
+ */
+function training_custom_access($role_name){
+  global $user;
+  return in_array($role_name, $user->roles);
+}
+
+/**
+ * Preprocessing argument. Returns the new value of the argument
+ * @param $arg - argument in URL training/menu/default_arg/%digit
+ * @return string
+ */
+
+function digit_load($arg){
+  switch ($arg) {
+    case 1:
+      $arg = 'one';
+      break;
+    case 2:
+      $arg = 'two';
+      break;
+    case 3:
+      $arg = 'three';
+      break;
+    case 99:
+      $arg = 'Almost there';
+      break;
+    default:
+      $arg = NULL;
+  }
+  return $arg;
+}
+
+/**
+ * Returns a dynamic page title
+ * string $user_name - contents user name
+ * @return string
+ */
+
+function training_menu_title_callback(){
+  global $user;
+  isset($user->name)?$user_name=$user->name:$user_name='anonymous';
+  return 'Dynamic title: username= '.$user_name;
+}
+
+
+/**
+ * Implements hook_permission()
+ * array $permission - array of permission
+ * @return array
+ */
+ function training_permission(){
+  $permission = array(
+    'train' => array(
+      'title' => 'Access protected menu example',
+      'description' => 'Use this to control access to page Permissioned Menu Item of training module',
+    ),
+    'train_theme' => array(
+      'title' => 'Access protected content',
+      'description' => 'Use this to control access to page Access Theme of training module',
+    ),
+  );
+
+  return $permission;
+}
+
+function training_google_form($form, &$form_state){
+  $form = array();
+  $form['group'] = array( //группа элементов формы
+    '#type' => 'fieldset',
+    '#title' => 'group',
+    '#collapsible' => TRUE, //сворачиваемость
+    '#collapsed' => TRUE, //свернута по умолчанию
+
+  );
+  $form['group']['text'] = array(
+    '#type' => 'textfield',
+    '#title' => 'title',
+  );
+  $form['group']['text2'] = array(
+    '#type' => 'textfield',
+    '#title' => 'title2',
+    '#required' => TRUE,
+  );
+  $form['#attached']['css'] = array(
+    //путь к файлу
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => 'ok'
+  );
+  return $form;
+}
+
+
+/**
+ * Implements hook_form_FORM_ID_alter
```

Comment:
Implements hook_form_FORM_ID_alter(). This is Drupal hook.

Remove all the rest.